### PR TITLE
PostCSS Stepped Value Functions

### PR DESCRIPTION
**Pull request by Antonio-Laguna**
Adds support for [Stepped Value Functions](https://www.w3.org/TR/css-values-4/#round-func) which are part of the [CSS Values and Units Module Level 4](https://www.w3.org/TR/css-values-4/).

This enables support for:

- `round()`
- `mod()`
- `rem()`

They were added to the spec in 2020 and little has changed since then. Because of that, it has been listed as [Stage 2](https://cssdb.org/#the-staging-process) on the CSSDB

### CODE REVIEW

File: plugins/postcss-stepped-value-functions/src/mod.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { Declaration, Result } from 'postcss';
```
Comment (romainmenke) [inline plugins/postcss-stepped-value-functions/src/mod.ts:1]:
```suggestion
import type { Declaration, Result } from 'postcss';
```

File: plugins/postcss-stepped-value-functions/src/rem.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { Declaration, Result } from 'postcss';
```
Comment (romainmenke) [inline plugins/postcss-stepped-value-functions/src/rem.ts:1]:
```suggestion
import type { Declaration, Result } from 'postcss';
```

File: plugins/postcss-stepped-value-functions/src/round.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
+import { Declaration, Result } from 'postcss';
```
Comment (romainmenke) [inline plugins/postcss-stepped-value-functions/src/round.ts:1]:
```suggestion
import type { Declaration, Result } from 'postcss';
```

File: plugins/postcss-stepped-value-functions/src/utils.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
+import valueParser, { FunctionNode, Node, WordNode } from 'postcss-value-parser';
+import { Declaration, Result } from 'postcss';
```
Comment (romainmenke) [inline plugins/postcss-stepped-value-functions/src/utils.ts:2]:
```suggestion
import valueParser from 'postcss-value-parser';
import type { FunctionNode, Node, WordNode } from 'postcss-value-parser';
import type { Declaration, Result } from 'postcss';
```

File: plugins/postcss-stepped-value-functions/test/basic.css [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+.invalid {
+	font-size: mod(18px, 5rem);
+	width: rem(var(--foo), 5px);
+	padding: mod(Fo0px, 5px);
+	left: mod(1px, 0px);
+	right: mod(1px, 200px);
+	inset: round(something, 15px, 4px);
+	margin-top: round(nearest, 15px, 4px, 50px);
+	margin-bottom: round(nearest, 15px, 4%);
+}
```
Comment (romainmenke) [inline plugins/postcss-stepped-value-functions/test/basic.css:42]: Should we add more tests for bogus things?

```css
font-size: mod;
font-size: mod();
font-size: mod(1px);
```

File: plugins/postcss-stepped-value-functions/test/basic.css [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+.rem {
+	padding: 8px rem(18px, 5px) 1px calc(rem(15px, 6px) + 50%);
+	transform: rotate(rem(-140deg, -90deg));
+}
+
+.mod-rem {
+	font-size: mod(-18px, 5px);
+	width: rem(-18px, 5px);
+	transform: rotateX(mod(140deg, -90deg)) rotateY(rem(140deg, -90deg));
+}
```
Comment (romainmenke) [inline plugins/postcss-stepped-value-functions/test/basic.css:15]: Maybe some more complex tests?

```css
font-size: round(nearest, rem(mod(mod(-18px, 5px), 5px), 2px));
```

nesting same function, combining all of them, ...

**Review by romainmenke — COMMENTED** [review]
Looks great!

Added a few notes :)

File: plugins/postcss-stepped-value-functions/docs/README.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+```
+
+Will produce on the terminal:
+
+```bash
+[postcss-stepped-value-functions]: Failed to transform mod(18px, 5rem) as the units don't match
+```
+
+<link-list>
+[CSS Specification]: https://www.w3.org/TR/css-values-4/#round-func
```
Comment (romainmenke) [inline plugins/postcss-stepped-value-functions/docs/README.md:77]:
```suggestion
[CSS Values 4]: <specUrl>
```

File: plugins/postcss-stepped-value-functions/docs/README.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+<!-- <example.css> file contents for examples/example.css -->
+<!-- <header> -->
+<!-- <usage> usage instructions -->
+<!-- <env-support> -->
+<!-- <link-list> -->
+<!-- to generate : npm run docs -->
+
+<header>
+
+[<humanReadableName>] lets you use `round`, `rem` and `mod` stepped value functions, following the [CSS Values 4][CSS Specification].
```
Comment (romainmenke) [inline plugins/postcss-stepped-value-functions/docs/README.md:17]:
```suggestion
[<humanReadableName>] lets you use `round`, `rem` and `mod` stepped value functions, following the [CSS Values 4].
```

File: plugins/postcss-stepped-value-functions/package.json [hunk 9]
```diff
@@ -0,0 +1,82 @@
+{
+	"name": "@csstools/postcss-stepped-value-functions",
+	"description": "TODO: Add description for Stepped Value Functions",
```
Comment (romainmenke) [inline plugins/postcss-stepped-value-functions/package.json:3]:
```suggestion
	"description": "Use round, rem and mod functions in CSS",
```

**Review by romainmenke — COMMENTED** [review]
Some small things, rest looks perfect!

**Comment by Antonio-Laguna** [comment]
@romainmenke thanks for catching those, will keep this in mind for the future!

**Review by romainmenke — APPROVED** [review]
LGTM!

Exited to get this out there 🎉

**Pull request merged**